go/mysql/binlog_dump.go

```diff
@@ -78,7 +78,15 @@ func (c *Conn) parseComBinlogDumpGTID(data []byte) (logFile string, logPos uint6
 		if !ok {
 			return logFile, logPos, position, readPacketErr
 		}
-		if gtid := string(data[pos : pos+int(dataSize)]); gtid != "" {
+
+		// NOTE: In testing a replication connection with a MySQL 8.0 replica, the replica sends back an 8 byte
+		//       GTID will all zero valued bytes, but Vitess was unable to parse it, so we added the nilGtid
```

Comment:
with all?

Reply:
I ended up changing this comment and the associated code – the MySQL docs for the replication protocol show that the GTID is sent as a binary GTID set, not as a human-readable string. I'm not sure why Vitess was trying to parse a human readable string here, but I suspect this code just isn't actually used by Vitess.